app/helpers/audits_helper.rb

```diff
@@ -257,9 +257,32 @@ def key_to_association_class(key, auditable_class)
     reflection_obj ? reflection_obj&.klass : nil
   end
 
+  def find_owner_class(key, change, audit)
+    type = nil
+    if audit.audited_changes.has_key?('owner_type')
+      type = audit.audited_changes['owner_type']
+      if audit.action == 'update'
+        idx = audit.audited_changes['owner_id'].index(change)
+        type = audit.audited_changes['owner_type'][idx]
+      end
+    else
+      previous = Audit.where(auditable_type: audit.auditable_type, auditable_id: audit.auditable_id)
+                   .where("id < ?", audit.id)
+                   .where("audited_changes LIKE '%owner_type:%'")
+                   .first
+      type = previous.audited_changes['owner_type']
```

Comment:
Also, this raised an exception for few times.

Reply:
when `previous` was `nil` I assume?